src/core/texpr.ml

```diff
@@ -568,11 +568,11 @@ let rec constructor_side_effects e =
 
 let type_constant basic c p =
 	match c with
-	| Int s ->
+	| Int (s,_) ->
```

Comment:
I don't think this function should ignore the suffixes.

Reply:
Yeah, wasn't sure what to do here since currently the suffixes are handled just before (mainly for the simplicity of creating the int64 make and uint cast with the untyped ast).

Would it be better to try and manually create the texprs here instead?

Reply:
Hmm, this situation makes me think that we should consider dealing with the `tconstant` part of this as well, but that would add some complexity to everything...

Generally, I haven't had great experiences with syntactic rewrites like this. However, I acknowledge that it's very convenient here.

Ultimately, we're going to need some sort of `mk_int64 : int64 -> texpr` function that doesn't require the full `typer` context. Which is already a problem because `Int64.make` is a pretty high-level function that relies on `inline`. I'll have to think about how to approach this.